src/test/java/org/prebid/server/bidder/sharethrough/SharethroughBidderTest.java

```diff
@@ -0,0 +1,283 @@
+
+    private static HttpCall<Void> givenHttpCall(Void bidRequest, String body) {
+        return HttpCall.success(
+                HttpRequest.<Void>builder().payload(bidRequest).build(),
+                HttpResponse.of(200, null, body),
+                null);
+    }
+
+    private static HttpCall<Void> givenHttpCallWithUri(String uri, String body) {
+        return HttpCall.success(
+                HttpRequest.<Void>builder().uri(uri).build(),
+                HttpResponse.of(200, null, body),
+                null);
+    }
+
+    @Before
+    public void setUp() {
+        sharethroughBidder = new SharethroughBidder(ENDPOINT_URL);
+    }
+
+    @Test
+    public void creationShouldFailOnInvalidEndpointUrl() {
+        assertThatIllegalArgumentException().isThrownBy(() -> new SharethroughBidder("invalid_url"));
+    }
+
+    @Test
+    public void makeHttpRequestsShouldReturnErrorWhenSiteIsNotPresent() {
+        // given
+        final BidRequest bidRequest = BidRequest.builder()
+                .imp(singletonList(Imp.builder()
+                        .ext(mapper.valueToTree(ExtPrebid.of(null, mapper.createArrayNode())))
+                        .build()))
+                .id("request_id")
+                .build();
+
+        // when
+        final Result<List<HttpRequest<Void>>> result = sharethroughBidder.makeHttpRequests(bidRequest);
+
+        // then
+        assertThat(result.getErrors()).hasSize(1);
+        assertThat(result.getErrors().get(0).getMessage()).startsWith("site.page is required");
+        assertThat(result.getValue()).isEmpty();
+    }
+
+    @Test
+    public void makeHttpRequestsShouldReturnErrorWhenImpExtCouldNotBeParsed() {
+        // given
+        final BidRequest bidRequest = BidRequest.builder()
+                .imp(singletonList(Imp.builder()
+                        .ext(mapper.valueToTree(ExtPrebid.of(null, mapper.createArrayNode())))
+                        .build()))
+                .id("request_id")
+                .site(Site.builder().page("http://page.com").build())
+                .build();
+
+        // when
+        final Result<List<HttpRequest<Void>>> result = sharethroughBidder.makeHttpRequests(bidRequest);
+
+        // then
+        assertThat(result.getErrors()).hasSize(1);
+        assertThat(result.getErrors().get(0).getMessage()).startsWith("Error occurred parsing sharethrough parameters");
+        assertThat(result.getValue()).isEmpty();
+    }
+
+    @Test
+    public void makeHttpRequestsShouldReturnRequestWithHeaderOriginDomainAndBodyNull() {
+        // given
+        final ArrayList<Integer> iframeSize = new ArrayList<>();
+        iframeSize.add(100);
+        iframeSize.add(200);
+        final BidRequest bidRequest = BidRequest.builder()
+                .imp(singletonList(Imp.builder()
+                        .ext(mapper.valueToTree(ExtPrebid.of(null,
+                                ExtImpSharethrough.of("pkey", false, iframeSize))))
```

Comment:
Instead of declaring, initializing and adding elements to `iframeSize` local variable, you can use `Arrays.asList` or even shorten it to `asList` (via static import) to declare a one-time-use ArrayList as follows: `asList(100, 200)`.

Also, you can use a `Collections.singletonList .singleton .singletonMap` whenever you need a collection with one element.